### DISCUSSION TITLE
Asking for code gen tool for graphql

### DISCUSSION THREAD

**Discussion by ZehuaZhang** [prisma/prisma#17854] [Q&A]
Is there any code gen tool to generate both graphql model / typeDef and graphql resolvers
I know there's typegraphql one, but I don't prefer the decorator syntax, which I think it is cumbersome and hard to read, will be a defactor later to maintain or scale.
Please correct me if Im wrong, graphql-codegen is only generating resolver, the mode/typeDef need to be defined again using pothos, so if there's any could generate both, that'd be great, please let me know, thank you.

**Comment by nurul3101** [comment]
Hello @ZehuaZhang 👋 

Did you have a chance to look at our [community generators](https://www.prisma.io/docs/concepts/components/prisma-schema/generators#community-generators) page? There are several generators listed there, I would recommend you to have a look once and see if anything's helpful to you.

**Reply by ZehuaZhang** [reply]
Hey there! Thanks for the link. From [community generators](https://www.prisma.io/docs/concepts/components/prisma-schema/generators#community-generators), the closest I got is the [prisma-generator-pothos-codegen](https://www.npmjs.com/package/prisma-generator-pothos-codegen) but is developed by one person and version is under 1, so I'm not sure I would use it

**Reply by nikolasburk** [reply]
Hey @ZehuaZhang 👋 there's also [Noun & Verb](https://nounandverb.io/) which generates a ready-made GraphQL API for you as well as the [generators from Pal.js](https://paljs.com/generator/) which also generate CRUD resolvers.

**Reply by ZehuaZhang** [reply]
@nikolasburk Thank you for those alternatives. [Noun & Verb](https://nounandverb.io/) is paid closed source, but [Pal.js](https://paljs.com/generator/) seems promising

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
